src/utils/selectors.ts

```diff
@@ -49,9 +49,20 @@ export const currentVenueSelectorData: SparkleSelector<AnyVenue | undefined> = (
   state
 ) => state.firestore.data.currentVenue;
 
-export const usersSelector: SparkleSelector<WithId<User>[]> = (state) =>
-  state.firestore.ordered.users ?? [];
+/**
+ * Selector to retrieve array of users from the Redux Firestore.
+ *
+ * @param state the Redux store
+ */
+export const usersSelector: SparkleSelector<WithId<User>[] | undefined> = (
+  state
+) => state.firestore.ordered.users;
```

Comment:
With this change, you seem to have removed the `?? []` that you added. Was that by explicit choice, or accidentally?

Reply:
By explicit choice because with `?? []` it didn't understand the status of this data. It thought it was loaded all the time. Since I have removed selectors everywhere, except hooks. I have ensured that we have this covered in the hooks

Reply:
Sounds good